DigitalLearningSolutions.Web/Views/Login/Index.cshtml

```diff
@@ -1,12 +1,82 @@
-﻿@{
+﻿@using DigitalLearningSolutions.Web.ViewModels.Login
+@model LoginViewModel 
+@{
   ViewData["Title"] = "Login";
+  var errorHasOccurred = !ViewData.ModelState.IsValid;
+  var usernameError = ViewData.ModelState["Username"]?.Errors?.Count > 0;
+  var passwordError = ViewData.ModelState["Password"]?.Errors?.Count > 0;
+  var passwordInputErrorClass = passwordError ? "nhsuk-input--error" : "";
+  var passwordFormErrorClass = passwordError ? "nhsuk-form-group--error" : "";
+  var usernameInputErrorClass = usernameError ? "nhsuk-input--error" : "";
+  var usernameFormGroupError = usernameError ? "nhsuk-form-group--error" : "";
 }
 
 <div class="nhsuk-grid-row">
   <div class="nhsuk-grid-column-full">
-    <h1 id="page-heading">Login</h1>
-    <a class="nhsuk-button" asp-controller="Login" asp-action="SignIn">
-      Sign in
+    <form method="post" novalidate asp-action="Login">
+
+      @if (errorHasOccurred) {
+        <partial name="_ErrorSummary"/>
+      }
+
+      <h1 id="page-heading">Login</h1>
+      <p>Enter your details to log into your account.</p>
+
+      <div class="nhsuk-form-group @usernameFormGroupError">
+        <label class="nhsuk-label" for="Username">
+          Email or user ID
+        </label>
+        @if (usernameError) {
+          <span class="nhsuk-error-message" id="username-error">
+            @foreach (var error in ViewData.ModelState["Username"].Errors) {
```

Comment:
Feels weird to get several username errors. The username either is not entered or is entered and doesn't exist in the database. Am I missing any error cases, or is this to satisfy error summary model?

Reply:
There should only ever be one username error, but I was trying to keep the error message production as general as possible. The error summary should not affect this. I can either :
- cut it back to just show the first error
- or move everything inside the "if" statement into a new partial view, keeping the "foreach" in case future use of the error message requires multiple errors to be displayed.

Reply:
I think unless the component is shared and needs being generic, I wouldn't make the logic more complicated than it needs to be as it will most likely be YAGNI. Hence, I think if it's straight-forward, we should only have one username error.